Frontend/src/Components/Dashboard/Dashboard.js

```diff
@@ -112,7 +116,15 @@ export default function Dashboard(props) {
       .then((res) => {
         //console.time("update react");
 
-        updateQueue(res.response);
+          // when the car/data generator is offline, res is going to be null,
+          //  so we need to set variables to a default value that will prevent a TypeError
+          //  when we try reading the properties of an undefined variable
+          if (res === null) {
+            // this object can't be null for the graph data, so we'll set it to a faux object
+            updateQueue(fauxQueue);
+          } else {
+            updateQueue(res.response);
+          }

```

Comment:
```suggestion
        // when the car/data generator is offline, res.response is going to be null,
        //  so we need to set variables to a default value that will prevent a TypeError
        //  when we try reading the properties of an undefined variable
        if (res.response === null) {
          // this object can't be null for the graph data, so we'll set it to a faux object
          updateQueue(fauxQueue);
        } else {
          updateQueue(res.response);
        }
```
Indentation is just a nit to make it consistent with the other lines. Changing `res` to `res.response` goes with the part of the review comment above that pertains to `api.js`